src/modules/navigator/navigator_main.cpp

```diff
@@ -307,25 +316,41 @@ Navigator::run()
 						}
 
 					} else if (PX4_ISFINITE(cmd.param7)) {
+						// Received only a request to change altitude, thus we keep the setpoint
+						rep->current.lat = curr->current.lat;
+						rep->current.lon = curr->current.lon;
+						rep->current.alt = cmd.param7;
+
+					} else {
+						// All three set to NaN - pause vehicle
+						rep->current.alt = get_global_position()->alt;
 
-						// Altitude without position change
-						// This condition is necessary for altitude changes just after takeoff where lat and lon are still nan
-						if (curr->current.valid && PX4_ISFINITE(curr->current.lat) && PX4_ISFINITE(curr->current.lon)) {
-							rep->current.lat = curr->current.lat;
-							rep->current.lon = curr->current.lon;
+						if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+						    && (get_position_setpoint_triplet()->current.type != position_setpoint_s::SETPOINT_TYPE_TAKEOFF)) {
+
+							// For multirotors we need to account for the braking distance, otherwise the vehicle will overshoot and go back
+							double lat, lon;
+							float course_over_ground = atan2f(_local_pos.vy, _local_pos.vx);
+
+							// predict braking distance
+
+							const float velocity_hor_abs = sqrtf(_local_pos.vx * _local_pos.vx + _local_pos.vy * _local_pos.vy);
+
+							float multirotor_braking_distance = math::trajectory::computeBrakingDistanceFromVelocity(velocity_hor_abs,
+											    _param_mpc_jerk_max, _param_mpc_acc_hor, 0.6f * _param_mpc_jerk_max);
```

Comment:
The jerk parameter used in auto modes is [MPC_JERK_AUTO](https://dev.px4.io/master/en/advanced/parameter_reference.html#MPC_JERK_AUTO)